src/gretel_trainer/models.py

```diff
@@ -0,0 +1,107 @@
+import logging
+from typing import Union
+
+from gretel_client.projects.models import read_model_config
+
+
+class _BaseConfig:
+    """This class should not be used directly, models should be
+    derived from this class
+    """
+
+    _max_header_clusters: int
+    """This should be overridden on concrete classes"""
+
+    # Default values for all configs, may be overridden if needed
+    _max_rows: int = 1000000
+    _enable_privacy_filters: bool = False
+
+    # Should be set by concrete constructors
+    config: Union[str, dict]
```

Comment:
Nit: `self.config` is always `dict`.
```suggestion
    config: dict
```

Reply:
In the constructor, a config can be a string- e.g. "synthetics/default" or a dictionary

Reply:
It can be a string only in the `__init__`, but this field will always be a `dict`, since it's initialized as `self.config = read_model_config(config)` ([link](https://github.com/gretelai/trainer/pull/12/files#diff-2319392ddaacefb2b7073a640aaa686428a65a968dc11262c1ad5ae94ae9d407R29)).

In the `__init__`, it's correctly annotated as `config: Union[str, dict]`.